### PR TITLE
No jetty

### PR DESCRIPTION
**Pull request by jaylu**
Hi Daniel, can helps to review ?

1. the jetty dependency is removed, using jdk http client.
2. all the tests are passed.
3. update the major verison to 1.0
4. the logging needs review before merging, so please don't merge it yet.

### CODE REVIEW

File: src/main/java/io/muserver/murp/ReverseProxyBuilder.java
Comment (danielflower) [inline src/main/java/io/muserver/murp/ReverseProxyBuilder.java:90]: If `true` and `ClientUtils.DISALLOWED__REQUEST_HEADERS` contains `host` then throw an exception telling the user to upgrade to Java 17 or set this to false.

Perhaps in the `build()` method just log a message noting which headers are disallowed (except for content-length)

File: src/main/java/io/muserver/murp/ReverseProxyBuilder.java
Comment (danielflower) [inline src/main/java/io/muserver/murp/ReverseProxyBuilder.java:200]: Always go secure-by-default.

This probably means you can delete the trust-all client from the source code (or maybe move it to the test source for testing). If people want to not trust certs then they'll need to put in the extra effort to achieve that.

File: src/test/java/io/muserver/murp/ReverseProxyTest.java
Comment (danielflower) [inline src/test/java/io/muserver/murp/ReverseProxyTest.java:48]: Maybe `ReverseProxyBuilder.createHttpClientBuilder()` should return this? (But with security enabled - or possibly allow the trust-all boolean to be passed in which would cancel my other comment about removing the trust-all code).

**Review by danielflower — CHANGES_REQUESTED** [review]
Nice work, really good to have jetty removed on this. Thanks.

File: src/main/java/io/muserver/murp/ReverseProxyBuilder.java
Comment (jaylu) [reply src/main/java/io/muserver/murp/ReverseProxyBuilder.java:90]: fixed and add log in build() method.

File: src/main/java/io/muserver/murp/ReverseProxyBuilder.java
Comment (jaylu) [reply src/main/java/io/muserver/murp/ReverseProxyBuilder.java:200]: will add ReverseProxyBuilder.createHttpCleint() as below, so that we can pass the trust all option to api user. 

`
    /**
     * Creates a new HTTP Client builder that is suitable for use in mu reverse proxy.
     *
     * @param trustAll If true, then any SSL certificate is allowed.
     * @return An HTTP Client builder
     */
    public static HttpClient.Builder createHttpClient(boolean trustAll) {
        return createHttpClientBuilder(trustAll)
            .followRedirects(HttpClient.Redirect.NEVER);
    }
`

File: src/test/java/io/muserver/murp/ReverseProxyTest.java
Comment (jaylu) [reply src/test/java/io/muserver/murp/ReverseProxyTest.java]: added ReverseProxyBuilder.createHttpClientBuilder

**Comment by jaylu** [comment]
@danielflower PR updated, can helps to take a second review ?

**Comment by jaylu** [comment]
update the logging, only log the error. normal logging will be controlled by the listener which implement by user.

**Comment by danielflower** [comment]
Nice one, thanks.

Can you add a file like https://github.com/3redronin/mu-server/blob/master/.github/workflows/ci.yaml (except Java 11, 17, and 21 if that's supported).

The title of this PR still says not to merge yet - I'm happy to merge this though if you're happy.

Thanks.

**Comment by jaylu** [comment]
FYI, I just try manual test with artillery, against ManualTest.java `/hi` end point which return 'hi' as body. 

Direct hit target server: 

```
--------------------------------
Summary report @ 17:10:48(+0800)
--------------------------------

http.codes.200: ................................................................ 2000
http.downloaded_bytes: ......................................................... 4000
http.request_rate: ............................................................. 1462/sec
http.requests: ................................................................. 2000
http.response_time:
  min: ......................................................................... 0
  max: ......................................................................... 36
  mean: ........................................................................ 4.2
  median: ...................................................................... 3
  p95: ......................................................................... 10.9
  p99: ......................................................................... 16.9
http.responses: ................................................................ 2000
vusers.completed: .............................................................. 20
vusers.created: ................................................................ 20
vusers.created_by_name.0: ...................................................... 20
vusers.failed: ................................................................. 0
vusers.session_length:
  min: ......................................................................... 198.1
  max: ......................................................................... 683.5
  mean: ........................................................................ 480.4
  median: ...................................................................... 528.6
  p95: ......................................................................... 658.6
  p99: ......................................................................... 658.6
```

hit via murp
```
--------------------------------
Summary report @ 17:10:57(+0800)
--------------------------------

http.codes.200: ................................................................ 2000
http.downloaded_bytes: ......................................................... 4000
http.request_rate: ............................................................. 782/sec
http.requests: ................................................................. 2000
http.response_time:
  min: ......................................................................... 0
  max: ......................................................................... 73
  mean: ........................................................................ 15.4
  median: ...................................................................... 13.9
  p95: ......................................................................... 30.9
  p99: ......................................................................... 44.3
http.responses: ................................................................ 2000
vusers.completed: .............................................................. 20
vusers.created: ................................................................ 20
vusers.created_by_name.0: ...................................................... 20
vusers.failed: ................................................................. 0
vusers.session_length:
  min: ......................................................................... 796.9
  max: ......................................................................... 1927.2
  mean: ........................................................................ 1663.7
  median: ...................................................................... 1755
  p95: ......................................................................... 1901.1
  p99: ......................................................................... 1901.1
```

**Comment by jaylu** [comment]
Thanks @danielflower , ci.yaml is added and I will merge it in directly.

**Comment by jaylu** [comment]
the first ci build failed, will raise another PR for it.

**Pull request merged**
